connect/mirror/src/test/java/org/apache/kafka/connect/mirror/MirrorConnectorsIntegrationTest.java

```diff
@@ -207,23 +212,45 @@ public void close() {
         backup.stop();
     }
 
+    // throw exception after 3 retries, and print expected error messages
+    private void assertEqualsWithConsumeRetries(final String errorMsg,
+                                                final int numRecordsProduces,
+                                                final int timeout,
+                                                final ClusterType clusterType,
+                                                final String... topics) throws InterruptedException {
+        int retries = 3;
+        while (retries-- > 0) {
+            try {
+                int actualNum = clusterType == ClusterType.PRIMARY ?
+                        primary.kafka().consume(numRecordsProduces, timeout, topics).count() :
```

Comment:
these are really strange side-effects to have an an assert statement. I see what you are trying to do, but this is probably not the way to do it.